test/unit/Ramm/getReserves.js

```diff
@@ -1,292 +1,236 @@
-  } else {
-    // extract eth
-    const elapsedLiquidity = elapsed.mul(LIQ_SPEED_A).mul(parseEther('1')).div(LIQ_SPEED_PERIOD);
-    const ethToTargetLiquidity = eth.sub(TARGET_LIQUIDITY);
-    const ethToExtract = elapsedLiquidity.lt(ethToTargetLiquidity) ? elapsedLiquidity : ethToTargetLiquidity;
-    eth = eth.sub(ethToExtract);
-  }
-
-  let nxmA = state.nxmA.mul(eth).div(state.eth);
-  let nxmB = state.nxmB.mul(eth).div(state.eth);
-
-  // apply ratchet below
-  // if cap*n*(1+r) > e*sup
-  // if cap*n.add(cap*n*r > e*sup
-  //   set n(new) = n(BV)
-  // else
-  //   set n(new) = n(R)
-  const r = elapsed.mul(state.ratchetSpeed);
-  const priceBufferA = PRICE_BUFFER_DENOMINATOR.add(PRICE_BUFFER);
-  const bufferedCapitalA = capital.mul(priceBufferA).div(PRICE_BUFFER_DENOMINATOR);
-
-  const kA = bufferedCapitalA.mul(nxmA);
-  const kARatchetValue = bufferedCapitalA.mul(nxmA).mul(r).div(RATCHET_PERIOD).div(RATCHET_DENOMINATOR);
-  const kARatcheted = kA.add(kARatchetValue);
-
-  if (kARatcheted.gt(eth.mul(supply))) {
-    // use bv
-    nxmA = eth.mul(supply).div(bufferedCapitalA);
-  } else {
-    // use ratchet
-    const nrDenomAddend = r.mul(capital).mul(nxmA).div(supply).div(RATCHET_PERIOD).div(RATCHET_DENOMINATOR);
-    nxmA = eth.mul(nxmA).div(eth.sub(nrDenomAddend));
-  }
-
-  // apply ratchet below
-  // check if we should be using the ratchet or the book value price using:
-  // Nbv > Nr <=>
-  // ... <=>
-  // cap * n < e * sup + r * cap * n
-  const priceBufferB = PRICE_BUFFER_DENOMINATOR - PRICE_BUFFER;
-  const bufferedCapitalB = capital.mul(priceBufferB).div(PRICE_BUFFER_DENOMINATOR);
-
-  const kB = bufferedCapitalB.mul(nxmB);
-  const kBRatchetValue = nxmB
-    .mul(capital)
-    .mul(elapsed)
-    .mul(state.ratchetSpeed)
-    .div(RATCHET_PERIOD)
-    .div(RATCHET_DENOMINATOR);
-  const kBRatcheted = eth.mul(supply).add(kBRatchetValue);
-  if (kB.lt(kBRatcheted)) {
-    // use bv
-    nxmB = eth.mul(supply).div(bufferedCapitalB);
-  } else {
-    // use ratchet
-    const nrDenomAddend = nxmB
-      .mul(elapsed)
-      .mul(state.ratchetSpeed)
-      .mul(capital)
-      .div(supply)
-      .div(RATCHET_PERIOD)
-      .div(RATCHET_DENOMINATOR);
-    nxmB = eth.mul(nxmB).div(eth.add(nrDenomAddend));
-  }
-
-  return {
-    nxmA,
-    nxmB,
-    eth,
-    budget,
-    ratchetSpeed: state.ratchetSpeed,
-    timestamp: currentTimestamp,
-  };
-};
-
-/* ========== Set Slo1.ethReserve helper functions ========== */
+const FAST_RATCHET_SPEED = 5000;
+const INITIAL_LIQUIDITY = parseEther('5000');
+const INITIAL_BUDGET = parseEther('43835');
+
+const getRammState = paramState => ({
+  nxmA: INITIAL_LIQUIDITY.mul(parseEther('1')).div(SPOT_PRICE_A),
+  nxmB: INITIAL_LIQUIDITY.mul(parseEther('1')).div(SPOT_PRICE_B),
+  eth: INITIAL_LIQUIDITY,
+  budget: INITIAL_BUDGET,
+  ratchetSpeed: FAST_RATCHET_SPEED,
+  timestamp: Math.floor(Date.now() / 1000),
+  ...paramState,
+});
```

Comment:
This function doesn't read from the actual state, it only returns the initial one, so the name is misleading. Except for the date everything else is pure so we can define it as a constant instead of a function. The date needs to be read from the chain as we cannot rely on the computer time (we may advance the time in evm so this will be inaccurate).